sway/desktop/output.c

```diff
@@ -385,9 +405,7 @@ static void begin_destroy(struct sway_output *output) {
 	output->wlr_output->data = NULL;
 	output->wlr_output = NULL;
 
-	transaction_commit_dirty();
-
-	update_output_manager_config(server);
+	request_modeset(server);
```

Comment:
Why do we want to modeset an output that we are in the process of destroying?

Reply:
We might be able to light up an output that the user wanted to light up but we were missing a CRTC for, I think?

Reply:
In that case, wouldn't we want to move the modeset out of `begin_destroy` and into the place where the output is actually destroyed?

Reply:
Yeah, we aren't modesetting *this* output, we're modesetting the entire backend in case this output was degrading other outputs (e.g., the usual bandwidth limits on Intel or not enough CRTCs).

> In that case, wouldn't we want to move the modeset out of begin_destroy and into the place where the output is actually destroyed?

`begin_destroy` seemed like the counterpart to `handle_new_output`, with them both ending in a similar epilogue - apply output configuration or evacuate, commit transaction, update `wlr_output_manager_v1`.

Just seemed easy to break both of those out into the same idle handler, and didn't think too much of it otherwise.